compiler/rustc_session/src/options.rs

```diff
@@ -58,10 +58,58 @@ macro_rules! hash_substruct {
     };
 }
 
+macro_rules! gather_tmods {
+    ($_opt_name:ident, $init:expr, $opt_expr:expr, $mods:expr, [SUBSTRUCT], [TMOD]) => {
+        compile_error!("SUBSTRUCT can't be TMOD (target modifier)");
+    };
+    ($opt_name:ident, $init:expr, $opt_expr:expr, $mods:expr, [UNTRACKED], [TMOD]) => {
+        if *$opt_expr != $init {
+            $mods.push(format!("{}={:?}", stringify!($opt_name), $opt_expr));
```

Comment:
We should avoid storing things like `Some(1)` or `CodeModel::Small` or w/e here and instead store whatever the user wrote, since that's what we'll want to show to them.

Reply:
Then, we probably need to save both technical representation for comparison and initial user-provided text for display. I think a little different user-provided option text may be parsed to the same technical value, right?
Like, `list_option=first,second` and `list_option=second,first`.

Reply:
I don't mind if the value is normalised in some way, like `first,second` becoming `second,first` as long as it matches what the user could write, unlike `Some(1)` or `CodeModel::Small`.